### ISSUE TITLE
lm模型参数怎么设置，除了bert数据需要什么其他的数据吗

### ISSUE THREAD

**Issue by cornorghost** [zjunlp/deepke#37]
启动lm模型找不到一些文件

**Comment by yuwl798180** [comment]
你好，readme中有有写 [常见问题](https://github.com/zjunlp/deepke#备注常见问题)，第九条，具体使用预训练语言模型，最好离线下载model以及相应的config等文件。

**Comment by cornorghost** [comment]
> 你好，readme中有有写 [常见问题](https://github.com/zjunlp/deepke#%E5%A4%87%E6%B3%A8%E5%B8%B8%E8%A7%81%E9%97%AE%E9%A2%98)，第九条，具体使用预训练语言模型，最好离线下载model以及相应的config等文件。

你好，感谢回复，在pretrained文件夹中的确按照要求放置了相应的三个文件，但是在运行时出现以下错误，他们看起来像是没有找到某些文件
`
[2020-01-04 22:16:48,581][preprocess][INFO] - convert relation into index...
[2020-01-04 22:16:48,588][preprocess][INFO] - verify whether use pretrained language models...
[2020-01-04 22:16:48,588][preprocess][INFO] - use pretrained language models serialize sentence...
[2020-01-04 22:16:48,589][preprocess][INFO] - use bert tokenizer...

[2020-01-04 22:16:48,589][transformers.tokenization_utils][INFO] - Model name 'pretrained' not found in model shortcut name list (bert-base-uncased, bert-large-uncased, bert-base-cased, bert-large-cased, bert-base-multilingual-uncased, bert-base-multilingual-cased, bert-base-chinese, bert-base-german-cased, bert-large-uncased-whole-word-masking, bert-large-cased-whole-word-masking, bert-large-uncased-whole-word-masking-finetuned-squad, bert-large-cased-whole-word-masking-finetuned-squad, bert-base-cased-finetuned-mrpc, bert-base-german-dbmdz-cased, bert-base-german-dbmdz-uncased, bert-base-finnish-cased-v1, bert-base-finnish-uncased-v1). Assuming 'pretrained' is a path or url to a directory containing tokenizer files.

[2020-01-04 22:16:48,591][transformers.tokenization_utils][INFO] - Didn't find file pretrained\added_tokens.json. We won't load it.

[2020-01-04 22:16:48,593][transformers.tokenization_utils][INFO] - Didn't find file pretrained\special_tokens_map.json. We won't load it.

[2020-01-04 22:16:48,594][transformers.tokenization_utils][INFO] - Didn't find file pretrained\tokenizer_config.json. We won't load it.

[2020-01-04 22:16:58,636][transformers.tokenization_utils][INFO] - loading file https://s3.amazonaws.com/models.huggingface.co/bert/pretrained/vocab.txt from cache at C:\Users\piguai\.cache\torch\transformers\f21cf6ad3df91b183c3c099e97ec33428878982268e28a5302bbb87d17771af7

[2020-01-04 22:16:58,636][transformers.tokenization_utils][INFO] - loading file None
[2020-01-04 22:16:58,636][transformers.tokenization_utils][INFO] - loading file None
[2020-01-04 22:16:58,637][transformers.tokenization_utils][INFO] - loading file None
`
![image](https://user-images.githubusercontent.com/33841855/71766959-17fc1e80-2f41-11ea-8fdf-736d0fd44b4c.png)

**Comment by yuwl798180** [comment]
你好，每一行后面都有提示解释的。

具体看下：
1. [2020-01-04 22:16:48,589][transformers.tokenization_utils][INFO]  这一行是说你提供的是本地的目录，所以预训练模型是采用离线的方式加载，不需要在线下载。 Assuming 'pretrained' is a path or url to a directory containing tokenizer files.

2. 下面几行是没有额外的 added_tokens， special_tokens_map，tokenizer_config文件，所以选择不加载。
前两个是为了解决：vocab文件里没有自己想要的词汇，可以自定义一些词汇表，加入到模型中，这样在训练过程中给新加入的词汇学出embedding。比如中文bert没有“箓”这个字，如果用 [unk] 表示可能影响模型效果。此时就可以自定义特殊token。bert自带的special有8个，比如[cls] [mask]等，如果你想添加任务相关的特殊token，比如[head_information]用来表示在学习过程中表示head entity整体的embedding效果。
第三个是token过程中的特殊配置，默认有一些基本配置，比如 uncased 表示不区分大小写，即全用小写模式，可以另外自定义一些。

这三个文件没有完全不影响模型运行的。

**Comment by cornorghost** [comment]
> 你好，每一行后面都有提示解释的。
> 
> 具体看下：
> 
> 1. [2020-01-04 22:16:48,589][transformers.tokenization_utils][INFO]  这一行是说你提供的是本地的目录，所以预训练模型是采用离线的方式加载，不需要在线下载。 Assuming 'pretrained' is a path or url to a directory containing tokenizer files.
> 2. 下面几行是没有额外的 added_tokens， special_tokens_map，tokenizer_config文件，所以选择不加载。
>    前两个是为了解决：vocab文件里没有自己想要的词汇，可以自定义一些词汇表，加入到模型中，这样在训练过程中给新加入的词汇学出embedding。比如中文bert没有“箓”这个字，如果用 [unk] 表示可能影响模型效果。此时就可以自定义特殊token。bert自带的special有8个，比如[cls] [mask]等，如果你想添加任务相关的特殊token，比如[head_information]用来表示在学习过程中表示head entity整体的embedding效果。
>    第三个是token过程中的特殊配置，默认有一些基本配置，比如 uncased 表示不区分大小写，即全用小写模式，可以另外自定义一些。
> 3. 下面3个 loading file None 就是说这三个文件没有找到，所以none。
> 
> 这三个文件没有完全不影响模型运行的。
> 如果对 transformers 有疑问，可以看下源码。

感谢如此详细的回答，我已经定位到出错的地方了，在lm.yaml配置文件里的
`lm_file:'pretrained'
`
改为绝对路径可以成功，我分析可能是我是用的编译器问题，我用的vs code，使用路径可能有问题，还是非常感谢，你的代码和回答给我很大启发和帮助
